File: src/test/java/org/vanautrui/languages/parsing/astnodes/nonterminals/TermNodeTest.java
package org.vanautrui.languages.parsing.astnodes.nonterminals;

import org.junit.Test;
import org.vanautrui.languages.lexing.utils.TokenList;
import org.vanautrui.languages.lexing.tokens.IdentifierToken;
import org.vanautrui.languages.lexing.tokens.IntegerNonNegativeConstantToken;
import org.vanautrui.languages.parsing.astnodes.nonterminal.TermNode;

public class TermNodeTest {

    @Test
    public void test_simple_term() throws Exception {

        TokenList list = new TokenList();
        list.add(new IntegerNonNegativeConstantToken(4));
        TermNode expr = new TermNode(list);
    }

    @Test
    public void test_variable_term() throws Exception {

        TokenList list = new TokenList();
        list.add(new IdentifierToken("x"));
        TermNode expr = new TermNode(list);
    }


}

File: dragon-parser/src/test/astnodes/nonterminals/TermNodeTest.cpp
package org.vanautrui.languages.test.astnodes.nonterminals;

import org.vanautrui.languages.compiler.lexing.tokens.IdentifierToken;
import org.vanautrui.languages.compiler.lexing.tokens.IntegerNonNegativeConstantToken;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.parsing.astnodes.nonterminal.TermNode;

import java.nio.file.Paths;

public final class TermNodeTest {

	public static int test_simple_term() {

		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));
			list.add(new IntegerNonNegativeConstantToken(4));
			final TermNode expr = new TermNode(list);
			return 1;
		}catch (Exception e){
			return 0;
		}
	}

	public static int test_variable_term() {

		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));
			list.add(new IdentifierToken("x"));
			final TermNode expr = new TermNode(list);
			return 1;
		}catch (Exception e){
			return 0;
		}
	}


}
